apps/publish/content/common.py

```diff
@@ -895,14 +895,31 @@ def _publish_associated_items(self, original, updates=None):
                         sync_associated_item_changes(associated_item, associated_item_updates)
                         continue
 
-                    if association_updates.get("state") not in PUBLISH_STATES or updates.get("state"):
+                    orig_associated_item = archive_service.find_one(req=None, _id=associated_item[config.ID_FIELD])
+                    if self.is_changed(orig_associated_item, associated_item):
                         remove_unwanted(association_updates)
                         publish_service.patch(id=associated_item[config.ID_FIELD], updates=association_updates)
 
             # When there is an associated item which is published, Inserts the latest version of that associated item into archive_versions.
             insert_into_versions(doc=associated_item)
         self._refresh_associated_items(original)
 
+    def is_changed(self, old: dict, new: dict) -> bool:
```

Comment:
we would need it also check the extra fields, and we now use profile for picture type so I would just look for any changes in fields not stating with `_`